base_layer/core/src/mempool/priority/prioritized_transaction.rs

```diff
@@ -88,6 +88,9 @@ impl PrioritizedTransaction {
         dependent_outputs: Option<Vec<HashOutput>>,
     ) -> Result<PrioritizedTransaction, TransactionError> {
         let weight = transaction.calculate_weight(weighting)?;
+        if weight == 0 {
+            return Err(TransactionError::ZeroWeight);
+        }
```

Comment:
It seems safer to replace the subsequent division by `weight` by a `checked_div`, which would make it much harder for a regression to sneak in. You could have failure of the `checked_div` return the same `TransactionError`  too, so the new test would still work the same.

Reply:
Good observation, thanks